feat: Update passive urls from passive.* to *-passive.*

robhemsley opened the pull request:
The following PR updates support for passive urls for deployed services. It switches from the previous naming convention of `passive.*` to `*-passive.*` in order to allow us to use our wild card SSL certs for each deployed environment `*.staging.` etc.

To support single host names we check the length of the split array and only concatenate if greater than 1.

edahlseng commented on sources/service-dns-cloudflare/main.tf:
```diff
@@ -22,10 +22,15 @@ resource "cloudflare_record" "active_domain" {
 resource "cloudflare_record" "passive_domain" {
-  for_each = local.passive_domains
+  for_each = {for y in local.passive_domains : y.name => {
+      host_tmp = split(".", y.name)
```
Can we rename this to make it easier to follow the intended logic flow?
```suggestion
      name_pieces = split(".", y.name)
```

edahlseng commented on sources/service-dns-cloudflare/main.tf:
```diff
@@ -22,10 +22,15 @@ resource "cloudflare_record" "active_domain" {
 
   zone_id = each.value.hosted_zone_id
-  name    = "passive.${each.value.name}"
+  name    = join(".", concat(["${substr(each.value.host_tmp[0], 0, 56)}-passive"], length(each.value.host_tmp) > 1 ? slice(each.value.host_tmp, 1, length(each.value.host_tmp)) : []))
```
Can we add a comment above this lines that clarifies why `56` is used as the substring length?

edahlseng commented on sources/service-kubernetes/main.tf:
```diff
@@ -32,10 +32,20 @@ locals {
     annotations = x.annotations
     rules = [for y in x.rules : {
-      host                        = "passive.${y.host}",
+      host_tmp                    = split(".", y.host),
```
Can we rename this to make the intended logic flow clearer?
```suggestion
      host_pieces                    = split(".", y.host),
```

edahlseng commented on sources/service-kubernetes/main.tf:
```diff
@@ -32,10 +32,20 @@ locals {
+  ingresses_passive = [for x in local.ingresses_passive_tmp : {
+    annotations = x.annotations
+    rules = [for y in x.rules : {
+      host                        = join(".", concat(["${substr(y.host_tmp[0], 0, 56)}-passive"], length(y.host_tmp) > 1 ? slice(y.host_tmp, 1, length(y.host_tmp)) : []))
```
Can we add a comment above this line that clarifies why `56` is used as the substring length?

edahlseng commented on sources/service-kubernetes/main.tf:
```diff
@@ -32,10 +32,20 @@ locals {
   docker_image_passive = var.docker_image_passive == "" ? local.docker_image_passive_calculated : var.docker_image_passive
 
-  ingresses_passive = [for x in var.ingresses : {
+  ingresses_passive_tmp = [for x in var.ingresses : {
```
I find `tmp` a little misleading here. Maybe we can consider something like `intermediate`, instead?
```suggestion
  ingresses_passive_intermediate = [for x in var.ingresses : {
```

edahlseng reviewed (changes requested):
The overall logic looks good! As currently written there are a number of design decisions that may be hard to follow in the future. I've left some inline comments with ways that we could make the intended logic a bit more clear. Additionally, let's strengthen up the commit message:
* For the commit message subject line, let's be more specific about what exactly is change. One example (feel free to modify from here) is something like `feat: Change passive hosts from passive.* to *-passive.*`. This should make the commit message a bit more glanceable.
* Can we add some additional context to the body of the commit message? It would be helpful to explain why this change is being made, why this approach is the chosen approach, and to clarify gotchas such as the use of the substring function for ensuring the host name falls under the maximum character limit.

robhemsley committed:
feat: Change passive host names from passive.* to *-passive.*

The following PR updates support for passive urls for deployed services. 
It switches from the previous naming convention of passive.* to 
*-passive.* in order to allow us to use our wild card SSL certs for each 
deployed environment *.staging. etc.

edahlseng reviewed (approved):
Tested over video call with @robhemsley. Looks good to merge!
